CMSgov/design-system
# [WNMGDS-1922] Update prop type definitions to support passthrough attributes

## Summary

The following components' prop type definitions were updated so that they allow additional attributes that will be passed through to an underlying HTML element:
- All the icon components
- `Badge`
- `SingleInputDateField`
- `Tooltip`

Note that this only satisfies the typing portion of this ticket. A follow-up PR will be made later.

## How to test

1. Icons
    - To test this, add `const addIcon = <AddIcon bbox="hello" />` to `SvgIcon.test.tsx` and run `yarn type-check` with and without the update
2. Badge, SingleInputDateField, Tootip
    - In `Badge.test.tsx`, try to add an attribute to `<Badge />` that is a valid HTML attribute but isn't in the props
    - Same kind of thing for the others

## Checklist
